sematic/config/server_settings.py

```diff
@@ -32,6 +37,9 @@ class ServerSettingsVar(AbstractSettingsVar):
     # GRAFANA
     GRAFANA_PANEL_URL = "GRAFANA_PANEL_URL"
 
+    # Server-side plug-ins section
+    plugins = "plugins"
```

Comment:
This entry denotes a different concept, and it creates validation and usability issues.

Given that we don't have plans to make the `default` profile configurable, it would align better if `plugins` would be a first-class citizen, just like profile:
```
default:
  SOME_SETTING_VAR: value
  OTHER_SETTING_VAR: other_value
plugins:
  - full.path.to.StoragePluginClass:
    - STORAGE_SETTING_VAR: value
  - full.path.to.MyCustomStoragePluginClass:
    - 
  - full.path.to.AuthPluginClass:
    - 
```

This way, we won't need to overload and complicate the profile's validation and handling.

In order to make the CLI integration work, each plugin should declare the settings that pertain to them. This can be done by having each plugin declare a field that extends the `SettingsScope`.

You would then be able to do both `sematic settings get SOME_SETTING_VAR` and `sematic settings get STORAGE_SETTING_VAR`, which will look in both sources and return the setting.

And `sematic settings set STORAGE_SETTING_VAR` will find the correct plugin section and save the setting there.

Reply:
This schema does not work. Plug-in settings are attached to profiles. You may have different e.g. BigQuery credentials for different profiles.

Having `plugins` as subkey to `default` is not a huge problem.

One thing I will change is to use the full import path as key in `settings` instead of just the plugin name, for more clarity.

Here are alternative schemas that would work well but require more changes and migrations:

Alternative A (namespacing top-level settings under settings)
```yaml
default:
  settings:
    SOME_SETTING_VAR: value
  plugins:
    scopes:
      STORAGE:
        - sematic.plugins.storage.S3Storage
    settings:
      sematic.plugins.storage.S3Storage:
        - AWS_S3_BUCKET: foobar
```

Alternative B (treat Sematic itself as a plug-in)
```yaml
default:
    scopes:
      SEMATIC:
        - sematic
      STORAGE:
        - sematic.plugins.storage.S3Storage
      AUTH:
        - sematic.plugins.auth.GoogleAuth
        - my_plugin_module.GitHubAuth
  settings:
    sematic:
      SOME_SETTING_VAR: value
    sematic.plugins.storage.S3Storage:
       AWS_S3_BUCKET: value
    sematic.plugins.auth.GoogleAuth:
       GOOGLE_OAUTH_CLIENT_ID: value
    my_plugin_module.GitHubAuth:
        GITHUB_OAUTH_CLIENT_ID: value
```

Alternative B would streamline the code since everything can be treated as a plug-in, but makes the settings file harder to read for humans.